incubator/hnc/pkg/controllers/object_controller.go

```diff
@@ -284,14 +291,31 @@ func (r *ObjectReconciler) onDelete(ctx context.Context, log logr.Logger, nnm ty
 	return nil
 }
 
+func (r *ObjectReconciler) ensureCanPropagate(nsNm string, inst *unstructured.Unstructured) bool {
+	r.Forest.Lock()
+	defer r.Forest.Unlock()
+	ns := r.Forest.Get(nsNm)
+	if !ns.Exists() {
+		return false
+	}
+	if inst.GetDeletionGracePeriodSeconds() != nil {
+		ns.SetCondition(fmt.Sprint(r.GVK, ns, nsNm), tenancy.ObjectInternal, "finalizer set")
+		return false
+	}
+	if inst.GetDeletionTimestamp() != nil {
+		ns.SetCondition(fmt.Sprint(r.GVK, ns, nsNm), tenancy.ObjectInternal, "finalizer set")
+		return false
+	}
+	if inst.GetFinalizers() != nil {
+		ns.SetCondition(fmt.Sprint(r.GVK, ns, nsNm), tenancy.ObjectInternal, "finalizer set")
+		return false
+	}
+	return true
+}
+
 func copyObject(inst *unstructured.Unstructured) *unstructured.Unstructured {
 	copied := inst.DeepCopy()
-
-	// Clear all irrelevant fields. cf https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#objectmeta-v1-meta
```

Comment:
Why delete this comment?

Reply:
Accidental, was writing lines around it and missed it in review.

Reply:
thx